### ISSUE TITLE
H2+ Support

### ISSUE THREAD

**Issue by scramblr** [jamesshao8/portapack-havoc-modified#18]
Hello, I was given an H2 to play with (not support them) and I see that your firmware does not support the H2 because they are closed source, etc - but will installing your code brick this device? I only ask because I do not want to brick my friend's device, but really wanted to test your version of havoc. 

Thank you

**Comment by jamesshao8** [comment]
There is no way to brick a device. You can always use DFU mode to recover the firmware running on HackRF in worst case.

But you won't find the latest apps from the code, as explained in ReadMe, they will only come with H3/H3 SE.

**Comment by antonzimin** [comment]
@jamesshao8 hello, sorry for disturbing you. I have **h2 PLUS** version of portapack. I have asked engeneers to get me related firmware. But they talked to me that it's not possible. Is it possible to use this havoc modified firmware for my device? Because currently it's works only with PC.

**Comment by jamesshao8** [comment]
> @jamesshao8 hello, sorry for disturbing you. I have **h2 PLUS** version of portapack. I have asked engeneers to get me related firmware. But they talked to me that it's not possible. Is it possible to use this havoc modified firmware for my device? Because currently it's works only with PC.

Hi there, my special firmware only support H3 and H3SE. It won't support H2plus even if H2plus looks similar as H3SE.

However, you may ask the seller to give you a firmware file of H2plus. According to my recollection, you will get battery voltage indicator from that.

**Comment by purplelaser456** [comment]
hello sorry i updateted my H3 se with the newest  nighthly build Firmware  i can Flash Back to Shaos Please help my. I can not find the Shaos Firmware

**Comment by jamesshao8** [comment]
> hello sorry i updateted my H3 se with the newest nighthly build Firmware i can Flash Back to Shaos Please help my. I can not find the Shaos Firmware

Don't worry. Use DFU mode to flash the firmware below.
[portapack-h3_with_add_on-shao.zip](https://github.com/jamesshao8/portapack-havoc-modified/files/9945321/portapack-h3_with_add_on-shao.zip)

To enter DFU mode, you can refer to wiki of HackRF.

**Comment by purplelaser456** [comment]
Thank You

Greez Dan

James Shao ***@***.***> schrieb am So., 6. Nov. 2022, 08:26:

**Comment by purplelaser456** [comment]
Hi James
You send my a bin file I must Copy to ?!? FOLDER.  I had before this File.
Greetings Dan
Sorry its New for My No Information on the GITHUB

Am So., 6. Nov. 2022 um 08:26 Uhr schrieb James Shao <
***@***.***>:

**Comment by jamesshao8** [comment]
Yes, it is a bin file. You need to somehow download it into hackrf’s flash. 

You can use dfu-util to write default firmware to hackrf’s ram. Then use hackrf_spiflash to download my bin file to hackrf’s rom/flash.

You can check detailed information on how other people unbrick a hackrf:
https://hackrf.readthedocs.io/en/latest/updating_firmware.html#only-if-necessary-dfu-boot

The only difference is to use my bin file for step 3.

PS: I highly recommend you to operate under Ubuntu instead of Windows.

**Comment by purplelaser456** [comment]
Sorry James im a looser i have only a Windows PC i think i can change the
bin File . I Have no Program for Windows  You have a Program  i can Load
the Bin. File and write to Hackrf. Iworkin 4 Hours.
 I Flash NEW Mayhem 1.54 Running now.
I have only this Programms See Pictures.
DFU MODE WORKING PERFECT I HAVE NO PROGRAM FOR READ AND WRITE THE BIN FILE.

Sorry
Greetings From Switzerland

Am Mo., 7. Nov. 2022 um 14:26 Uhr schrieb Daniel Ott ***@***.***>:

**Comment by jamesshao8** [comment]
> Sorry James im a looser i have only a Windows PC i think i can change the bin File . I Have no Program for Windows You have a Program i can Load the Bin. File and write to Hackrf. Iworkin 4 Hours. I Flash NEW Mayhem 1.54 Running now. I have only this Programms See Pictures. DFU MODE WORKING PERFECT I HAVE NO PROGRAM FOR READ AND WRITE THE BIN FILE. Sorry Greetings From Switzerland Am Mo., 7. Nov. 2022 um 14:26 Uhr schrieb Daniel Ott ***@***.***>:
> […](#)

Hi Daniel,


I already got your email. Will guide you through that.
